### PR TITLE
chore(views): Adds elgg_view_resource() for rendering full-page views

### LINKED ISSUE #8436: Resource views broke alternate viewtype renderings (RSS)
Somehow resource views need to default to the "default" viewtype instead of the global viewtype. @ewinslow

### PR DESCRIPTION
**Pull request by mrclay**
This makes sure that resource views are requested with the “default” viewtype and issues an error and 404 if the view is missing.

Fixes #8436

### CODE REVIEW

**Comment by mrclay** [comment]
Updated with logic discussed in #8436

File: docs/examples/plugins/start.php [hunk 1]
```diff
@@ -14,7 +14,10 @@ function my_plugin_init() {
 	elgg_register_action('my_plugin', "$base_dir/my_action.php");
 
 	// Extend the main CSS file
-	elgg_extend_view('css/elgg', 'my_plugin/css');
+	elgg_extend_view('css/elgg', 'css/my_plugin.css');
+
+	// Require your JavaScript AMD module (view "js/my_plugin.js") on every page
+	elgg_require_js('my_plugin');
```
Comment (ewinslow) [inline docs/examples/plugins/start.php]: These top two changes are great but seem unrelated to this PR.

File: docs/guides/guidelines.rst [hunk 2]
```diff
@@ -40,7 +40,7 @@ Use standardized routing with page handlers
 - Include page handler scripts from the page handler. Almost every page handler should have a page handler script. (Example: ``bookmarks/all`` => ``mod/bookmarks/pages/bookmarks/all.php``)
 - Call ``set_input()`` for entity guids in the page handler and use ``get_input()`` in the page handler scripts.
 - Call ``elgg_gatekeeper()`` and ``elgg_admin_gatekeeper()`` in the page handler function if required.
-- The group URL should use the ``pages/<handler>/owner.php`` script.
+- The group URL should use views like ``resources/groups/*.php`` to render pages.
```
Comment (ewinslow) [inline docs/guides/guidelines.rst]: +1 thank you

File: docs/guides/guidelines.rst [hunk 3]
```diff
@@ -40,7 +40,7 @@ Use standardized routing with page handlers
 - Include page handler scripts from the page handler. Almost every page handler should have a page handler script. (Example: ``bookmarks/all`` => ``mod/bookmarks/pages/bookmarks/all.php``)
```
Comment (ewinslow) [inline docs/guides/guidelines.rst]: Does this line need to change too?

File: docs/guides/plugins/plugin-skeleton.rst [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Plugin skeleton
```
Comment (ewinslow) [inline docs/guides/plugins/plugin-skeleton.rst]: Thank you, thank you, thank you for updating this file.

File: docs/guides/plugins/plugin-skeleton.rst [hunk 5]
```diff
@@ -15,35 +15,28 @@ The following files for plugin ``example`` would go in ``/mod/example/``
     actions/
         example/
             action.php
-        other_action.php
+            other_action.php
     classes/
-        ExampleClass.php
+        VendorNamespace/
+            ExampleClass.php
     graphics/
```
Comment (ewinslow) [inline docs/guides/plugins/plugin-skeleton.rst]: Can we move this into views now?

File: docs/tutorials/hello_world.rst [hunk 6]
```diff
@@ -87,6 +87,8 @@ Update the ``start.php`` to look like this:
     }
     
     function hello_world_page_handler() {
+    	echo elgg_view_resource('hello');
+
     	$params = array(
```
Comment (ewinslow) [inline docs/tutorials/hello_world.rst]: You added elgg_view_resource above, but didn't delete the params and elgg_view_page below. Seems odd.

**Comment by ewinslow** [comment]
I was skeptical at first but I actually like that this formalizes the convention of "resources/*"

**Comment by mrclay** [comment]
Updated

File: docs/guides/guidelines.rst (hunk 3 above)
Comment (ewinslow) [inline docs/guides/guidelines.rst]: Looks like it does. Can you update it?

Comment (mrclay) [inline docs/guides/guidelines.rst]: Done

**Comment by ewinslow** [comment]
LGTM
